app/src/features/apiClient/screens/apiClient/components/clientView/components/request/RequestBody.tsx

```diff
@@ -37,6 +38,13 @@ const RequestBody: React.FC<Props> = ({ body, contentType, variables, setRequest
     setRequestEntry((prev) => ({ ...prev, request: { ...prev.request, body: updatedBody } }));
   }, [contentType, rawBody, jsonBody, formBody, setRequestEntry]);
 
+  useEffect(() => {
+    if (contentType === RequestContentType.JSON && typeof jsonBody === "string") {
+      const prettifiedContent = prettifyCode(jsonBody, EditorLanguage.JSON).code;
+      setJsonBody(prettifiedContent);
+    }
+  }, [contentType]);
+
```

Comment:
Why is this required?

Reply:
Issue: when switching between RAW & json, the prettification gets removed
Solution: added a useEffect on json content type, which will prettify when switching the content type to JSON